actions/userinfo.js

```diff
@@ -1,17 +1,218 @@
+import Axios from 'axios';
 import local from './axiosConfigInitial';
 import { browserHistory } from 'react-router';
 import { error } from './error';
+import { AUTH_CONFIG } from './auth0-variables';
 
 export const SUBMIT_USER_INFO_BEGIN = 'SUBMIT_USER_INFO_BEGIN';
 export const SUBMIT_USER_INFO_SUCCESS = 'SUBMIT_USER_INFO_SUCCESS';
 export const SUBMIT_COMMENTS_BEGIN = 'SUBMIT_COMMENTS_BEGIN';
 export const SUBMIT_COMMENTS_SUCCESS = 'SUBMIT_COMMENTS_SUCCESS';
 export const USER_ID = 'USER_ID';
+export const LOGIN_REQUEST = 'LOGIN_REQUEST';
+export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
+export const LOGIN_ERROR = 'LOGIN_ERROR';
+export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
+export const GENERATE_QUIZ_USER = 'GENERATE_QUIZ_USER';
+export const LOGIN_LOCATION = 'LOGIN_LOCATION';
 
-function sendUserId(id) {
+const lock = new Auth0Lock(AUTH_CONFIG.clientId, AUTH_CONFIG.domain, {
+  languageDictionary: {
+    title: 'Pushkin'
+  },
+  // allowedConnections: ['facebook', 'Username-Password-Authentication'],
+  // oidcConformant: true,
+  autoclose: true,
+  autoParseHash: true,
+  auth: {
+    autoParseHash: true,
+    // connectionScopes: {
+    //   facebook: ['email']
+    // },
+    // redirectUrl: AUTH_CONFIG.callbackUrl,
+    // redirectUrl: 'http://localhost:8000/loading',
+    redirect: false,
+    // popup: true,
+    // sso: true,
+    responseType: 'token id_token',
+    // audience: AUTH_CONFIG.audience,
+
+    params: {
+      scope: 'openid email' // Learn about scopes: https://auth0.com/docs/scopes
+      // state: 36000
+    }
+  }
+});
+export function login() {
+  lock.show();
+  return dispatch => {
+    return dispatch(loginRequest());
+  };
+}
+function loginRequest() {
+  return {
+    type: LOGIN_REQUEST
+  };
+}
+
+export function loginSuccess(profile) {
+  return {
+    type: LOGIN_SUCCESS,
+    profile
+  };
+}
+
+function loginError(error) {
+  return {
+    type: LOGIN_ERROR,
+    error
+  };
+}
+export function loginLocation(location) {
+  return {
+    type: 'LOGIN_LOCATION',
+    location
+  };
+}
+export function checkLogin(location) {
+  return (dispatch, getState) => {
+    const location = getState().userInfo.loginLocation;
+    // Add callback for lock's `authenticated` event
+    lock.on('authenticated', authResult => {
+      return new Promise((res, rej) => {
+        res(authResult);
+      })
+        .then(authResult => {
+          let expiresAt = JSON.stringify(36000 * 1000 + new Date().getTime());
```

Comment:
I like to write these out so they're easily understandable: `24 * 60 * 60 * 1000` = 24 hours

Reply:
check